pkg/task/task.go

```diff
@@ -0,0 +1,197 @@
+/*
```

Comment:
Not clear why this functionality was moved under /pkg. Any code in /pkg cannot have dependencies on anything in the snap framework.